### ISSUE TITLE
A netstring carries size information. It is encoded as 4-byte uint32. Why?

### ISSUE THREAD

**Issue by seandlg** [kyrylo/netstring#4]
From `netstring.go` it is claimed that:
> // A netstring carries size information. It is encoded as 4-byte uint32

This leaves me confused. From [Bernstein's website](https://cr.yp.to/proto/netstrings.txt), it clearly states that:

> Any string of 8-bit bytes may be encoded as [len]":"[string]",".
   Here [string] is the string and [len] is a nonempty sequence of ASCII
   digits giving the length of [string] in decimal

Am I overseeing something here? Why is the string's length not interpreted dynamically from the data stream?

**Comment by kyrylo** [comment]
Hey there!

I think I tried to cater this library to my personal needs when I wrote it. At the time it was convenient for me to use `uint32` but I cannot remember why. Since I no longer use this and I am not aware of any users of this library, it makes sense to change this to `uint64` and make a major version bump. Will that work for you?

**Comment by seandlg** [comment]
Thanks for the reply! Actually it's a little more complicated than that, since the length is `ASCII`-encoded digits of arbitrary length. If I find time I will look into this an open a PR :)

Something along the lines of (as seen [here](https://github.com/paypal/hera/blob/master/utility/encoding/netstring/netstring.go)):

```go
var buff bytes.Buffer
var tmp = make([]byte, 1)
var digit int
var err error
// read length
length := 0
for {
    _, err = _reader.Read(tmp)
    b := tmp[0]
    if err != nil {
        return nil, err
    }
    buff.WriteByte(b)
    if b == colon {
        break
    } else {
        digit = int(b - '0')
        if (digit < 0) || (digit > 9) {
            return nil, errors.New("expected digit reading length")
        }
        length = length*10 + digit
    }
}
```

**Comment by kyrylo** [comment]
Nice find, I wasn't aware of that library. That's an interesting approach. Nevertheless, a PR that improves this repo is highly welcome. 🤝

I guess this library will probably fail to calculate the correct length for netstrings longer than `4294967295` (max uint32 value). Perhaps, my assumption was that this was more than enough.

**Comment by Natureshadow** [comment]
Ehm, maybe you could at least update the readme, which references djb's protocol, and clearly state that this library does **not** implement this protocol? If you don't want to fix it, ok, but please clearly state that it is broken (or implements your own standard rather than that of djb.

**Comment by kyrylo** [comment]
I just released v2.0.0 which does implement the spec and fixes the length issue. Thanks again!
